### PR TITLE
Add jcstress test to system test

### PR DESCRIPTION
**Pull request by Mesbah-Alam**
Related : https://github.com/adoptium/aqa-tests/issues/3546

Should be delivered after: https://github.com/adoptium/TKG/pull/321

Signed-off-by: Mesbah-Alam <Mesbah_Alam@ca.ibm.com>

### CODE REVIEW

**Comment by Mesbah-Alam** [comment]
Test : Grinder/23850

File: system/jcstress/playlist.xml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+-->
+<playlist xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../../TKG/playlist.xsd">
+	<test>
+		<testCaseName>jcstress_SampleTestBench</testCaseName>
+		<command>$(JAVA_COMMAND) $(JVM_OPTIONS) -jar $(Q)${TEST_ROOT}$(D)systemtest_prereqs$(D)jcstress$(D)jcstress-latest.jar$(Q) -jvmArgs $(Q)$(JVM_OPTIONS) -t SampleTestBench$(Q); \
+		$(TEST_STATUS)</command>
+		<versions>
+			<version>11</version>
```
Comment (llxia) [inline system/jcstress/playlist.xml:21]: Should this be 8+? If so, we do not need the version tag.

Comment (Mesbah-Alam) [reply system/jcstress/playlist.xml]: Good catch. Updated, works. Grinder/23855.

**Comment by llxia** [comment]
Ideally, we should separate jcstress with STF as they are unrelated. However, this PR creates the dependency between jcstress and STF (i.e, we cannot run jcstress without STF). And without build.xml, I suspect dynamic compilation will not work in this case.

I am ok with the PR for now as this enables us to run jcstress. Please open an issue for decoupling jcstress and STF. I think the better way is to use getDependencies in TKG.

**Comment by Mesbah-Alam** [comment]
> we cannot run jcstress without STF

We can run jcstress without STF. The only dependency is the top.xml changes (https://github.com/adoptium/STF/pull/127). However, since we want to run jckstress as part of systemtest, and system test will compile STF anyway, there is no extra cost of having the dependency on top.xml to fetch jcstress.jar.

**Comment by smlambert** [comment]
Ideally for the dynamic compilation feature that TKG has, if I only wanted to run the jcstress target by itself, then it should not have to go and compile STF but agree any further decoupling can be in a 'phase 2'.

**Comment by Mesbah-Alam** [comment]
Sounds good! Instead of doing it in phase 2 , let's move to using the getDependencies script in TKG now. 

I've added jcstress jar to the list of downloads in get Dependencies script in TKG : https://github.com/adoptium/TKG/pull/321. 

Manually ran test.getDependency job pointing to my branch for testing : https://hyc-runtimes-jenkins.swg-devops.com/job/test.getDependency/10261/. 

(Note, once https://github.com/adoptium/TKG/pull/321 is delivered, we need to run the [test.getDependency job at ci.adopt](https://ci.adoptopenjdk.net/job/test.getDependency/)). 

Updated the PR for playlist. New test : Grinder/23939.

**Comment by llxia** [comment]
Thanks @Mesbah-Alam . 

Please also add build.xml file in this case. Otherwise, I think the local run will fail. :) 
The build.xml should contain the following. It will trigger getDependencies for downloading jcstress if needed.
```
	<property name="LIB" value="jcstress" />
	<import file="${TEST_ROOT}/TKG/scripts/getDependencies.xml" />
```

**Comment by Mesbah-Alam** [comment]
Re: https://github.com/adoptium/aqa-tests/pull/3666#issuecomment-1130404099

Added. Please check if this will suffice!

File: system/jcstress/build.xml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+-->
+
+<project name="JCStressTest" basedir=".">
+	<property name="LIB" value="jcstress" />
+	<import file="${TEST_ROOT}/TKG/scripts/getDependencies.xml" />
+</project>
```
Comment (llxia) [inline system/jcstress/build.xml:20]: Sorry, the above is not sufficient. We need the default targets.

```
<project name="JCStressTest" default="build" basedir=".">
	<taskdef resource="net/sf/antcontrib/antlib.xml" />
	<description>JCStress Test</description>

	<!-- Set global properties for this build -->
	<property name="DEST" value="${BUILD_ROOT}/system/jcstress" />
	<property name="SRC" location="." />

	<property name="LIB" value="jcstress" />
	<import file="${TEST_ROOT}/TKG/scripts/getDependencies.xml" />

	<target name="init">
		<mkdir dir="${DEST}" />
	</target>
	
	<target name="dist" depends="init" description="generate the distribution">
		<copy todir="${DEST}">
			<fileset dir="${SRC}"/>
		</copy>
	</target>

	<target name="build">
		<antcall target="dist" inheritall="true" />
	</target>
</project>
```

Comment (llxia) [reply system/jcstress/build.xml:20]: This should be tested via `BUILD_LIST=system/jcstress`

Comment (Mesbah-Alam) [reply system/jcstress/build.xml:41]: Added (tested: Grinder/23946). Please check.

Comment (Mesbah-Alam) [reply system/jcstress/build.xml:41]: Here's the test with `BUILD_LIST=system/jcstress` : Grinder/24008.

**Review by llxia — APPROVED** [review]
LGTM

**Review by smlambert — APPROVED** [review]
Since the related TKG PR is merged, I will proceed to merge this one.  Thanks @Mesbah-Alam for this initial add, and as discussed, we can now refine what we run out of the jcstress repo quite easily in a follow-up PR. 👍

**Pull request merged**